osu.Game/Rulesets/Edit/Checks/CheckDrainTime.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Edit.Checks.Components;
+
+namespace osu.Game.Rulesets.Edit.Checks
+{
+    public class CheckDrainTime : ICheck
+    {
+        private const int min_drain_threshold = 30 * 1000;
+        public CheckMetadata Metadata => new CheckMetadata(CheckCategory.Compose, "Too short drain time");
+
+        public IEnumerable<IssueTemplate> PossibleTemplates => new IssueTemplate[]
+        {
+            new IssueTemplateTooShort(this)
+        };
+
+        public IEnumerable<Issue> Run(BeatmapVerifierContext context)
+        {
+            double drainTime = context.Beatmap.CalculatePlayableLength();
```

Comment:
This is not what drain time is. Drain time is, as the name suggests, the amount of time during gameplay wherein HP drain is active. HP drain is not active during breaks. `CalculatePlayableLength()` does not account for breaks in any way, shape, or form.

Compare implementation in [Mapset](https://github.com/Naxesss/MapsetChecks/blob/8efbf76d17aa495732b2de0b75e33132169cecd4/Checks/AllModes/Compose/CheckDrainTime.cs#L55) [Verifier](https://github.com/Naxesss/MapsetParser/blob/8f14354073ed9f2c4723bbd96ac8602cdc6693b6/objects/Beatmap.cs#L606).

Reply:
You are right about this, apologies for the oversight. I will get back to adressing the feeeback soon, but I assume that drain time is not currently calculated (or at least I cannot find the relevant code for now). Should the calculations also be added to beatmap extensions in case they become reused for future UI?